File: FileHelpers.CodeExamples/Example VbNet/CustomerFixedLength.vb

'-> Never forget:  Imports FileHelpers
Imports FileHelpers


<FixedLengthRecord()> _
Public Class Customer2

    <FieldFixedLength(5)> _
    Public CustId As Integer

    <FieldFixedLength(20), _
     FieldTrimAttribute(TrimMode.Right)> _
    Public Name As String

    <FieldFixedLength(8), _
     FieldConverter(GetType(TwoDecimalConverter))> _
    Public Balance As Decimal

    <FieldFixedLength(8), _
    FieldConverter(ConverterKind.Date, "ddMMyyyy")> _
    Public AddedDate As DateTime

End Class

Friend Class TwoDecimalConverter
    Inherits ConverterBase

    Public Overrides Function StringToField(ByVal from As String) As Object
        Dim res As Decimal = Convert.ToDecimal(from)
        Return res / 100
    End Function

    Public Overrides Function FieldToString(ByVal from As Object) As String
        Dim d As Decimal = CType(from, Decimal)
        Return Math.Round(d * 100).ToString()
    End Function

End Class

File: FileHelpers.CodeExamples/Example CSharp - VS 2005/CustomerFixedLength.cs
using System;
using FileHelpers;

namespace Examples
{
	[FixedLengthRecord()]
	public class Customer2
	{
		[FieldFixedLength(5)]
		public int CustId;
		
		[FieldFixedLength(20)]
		[FieldTrim(TrimMode.Right)]
		public string Name;

		[FieldFixedLength(8)]
		[FieldConverter(typeof(TwoDecimalConverter))]
		public decimal Balance;

		[FieldFixedLength(8)]
		[FieldConverter(ConverterKind.Date, "ddMMyyyy")]
		public DateTime AddedDate;

	}

	#region "  Decimal Converter  "

	internal class TwoDecimalConverter: ConverterBase
	{
		public override object StringToField(string from)
		{
			decimal res = Convert.ToDecimal(from);
			return res / 100;
		}

		public override string FieldToString(object from)
		{
			decimal d = (decimal) from;
			return Math.Round(d * 100).ToString();
		}
	
	}

	#endregion

}
